llm/openai/openai.go

```diff
@@ -221,25 +239,37 @@ func (o *OpenAI) stream(
 
 	var messages []*thread.Message
 	var content string
+	var allToolCalls []openai.ToolCall
```

Comment:
To reduce complexity we have to put 242-278 to a separated method.

Reply:
@henomis , I tried to move the entire block and tried running `golangci-lint` but it still threw the same error. 
I have thus moved only the error handling logic to a separate function to reduce complexity. 
`golangci-lint` doesnt throw any errors now.

Reply:
Cool! Let me reserve a slot of time for the last review, and we should be ready to merge it. Thank you so much for your effort! :rocket: